selfservice/flow/login/handler.go

```diff
@@ -110,6 +111,12 @@ func (h *Handler) NewLoginRequest(w http.ResponseWriter, r *http.Request, redir
 //       500: genericError
 func (h *Handler) initLoginRequest(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	if err := h.NewLoginRequest(w, r, func(a *Request) (string, error) {
+		// we assume an error means the user has no session
+		if _, err := h.d.SessionManager().FetchFromRequest(r.Context(), w, r); err == nil {
+			if err := h.d.LoginRequestPersister().UpdateLoginRequestReauth(r.Context(), a.ID, true); err != nil {
```

Comment:
This implies that if you're trying to create a new login request, you will always end up with a login flow. The idea of the reauth functionality however is that we can tell kratos that we want reauthentication, regardless of the reauth state. This could be achieved by, for example, calling:

```
/self-service/browser/flows/login?prompt=true
```